src/test/java/org/jenkinsci/plugins/github_branch_source/GithubSCMSourceTagsTest.java

```diff
@@ -0,0 +1,443 @@
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.kohsuke.github.GHFileNotFoundException;
+import org.kohsuke.github.GHException;
+import org.kohsuke.github.GHRef;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+import org.kohsuke.github.PagedIterator;
+import org.kohsuke.github.PagedIterable;
+import org.mockito.Mockito;
+
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static org.junit.Assert.*;
+
+@RunWith(Parameterized.class)
+public class GithubSCMSourceTagsTest {
+
+    /**
+     * All tests in this class only use Jenkins for the extensions
+     */
+    @ClassRule
+    public static JenkinsRule r = new JenkinsRule();
+
+    public static WireMockRuleFactory factory = new WireMockRuleFactory();
+
+    @Rule
+    public WireMockRule githubRaw = factory.getRule(WireMockConfiguration.options()
+            .dynamicPort()
+            .usingFilesUnderClasspath("raw")
+    );
+    @Rule
+    public WireMockRule githubApi = factory.getRule(WireMockConfiguration.options()
+            .dynamicPort()
+            .usingFilesUnderClasspath("api")
+            .extensions(
+                    new ResponseTransformer() {
+                        @Override
+                        public Response transform(Request request, Response response, FileSource files,
+                                                  Parameters parameters) {
+                            if ("application/json"
+                                    .equals(response.getHeaders().getContentTypeHeader().mimeTypePart())) {
+                                return Response.Builder.like(response)
+                                        .but()
+                                        .body(response.getBodyAsString()
+                                                .replace("https://api.github.com/",
+                                                        "http://localhost:" + githubApi.port() + "/")
+                                                .replace("https://raw.githubusercontent.com/",
+                                                        "http://localhost:" + githubRaw.port() + "/")
+                                        )
+                                        .build();
+                            }
+                            return response;
+                        }
+
+                        @Override
+                        public String getName() {
+                            return "url-rewrite";
+                        }
+
+                    })
+    );
+    private GitHubSCMSource source;
+    private GitHub github;
+    private GHRepository repo;
+
+    public GithubSCMSourceTagsTest(GitHubSCMSource source) {
+        this.source = source;
+    }
+
+    @Parameterized.Parameters(name = "{index}: revision={0}")
```

Comment:
remove

Reply:
It threw java.lang.Exception: No public static parameters method on class org.jenkinsci.plugins.github_branch_source.GithubSCMSourceTagsTest